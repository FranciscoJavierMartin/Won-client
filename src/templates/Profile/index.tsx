import { useRouter } from 'next/router';
import Base from '@/templates/Base';
import { Container } from '@/components/Container';
import Heading from '@/components/Heading';
import ProfileMenu from '@/components/ProfileMenu';
import * as S from './styles';

export type ProfileProps = {
  children: React.ReactNode;
};

const Profile: React.FC<ProfileProps> = ({ children }) => {
  const { asPath } = useRouter();
  return (
    <Base>
      <Container>
        <Heading lineLeft lineColor="secondary">
          My profile
        </Heading>
        <S.Main>
          <ProfileMenu activeLink={asPath} />
          <S.Content>{children}</S.Content>
        </S.Main>
      </Container>
    </Base>
  );
};
export default Profile;
