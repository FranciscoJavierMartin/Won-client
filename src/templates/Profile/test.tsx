import { screen } from '@testing-library/react';
import { renderWithTheme } from '@/utils/tests/helpers';
import Profile from '.';

describe('<Profile />', () => {
  xit('should render component', () => {
    const { container } = renderWithTheme(<Profile />);

    expect(
      screen.getByRole('heading', { name: /Profile/i })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
