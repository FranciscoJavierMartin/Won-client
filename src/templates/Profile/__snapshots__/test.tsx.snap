// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> should render sections 1`] = `
.c0 {
  width: 100%;
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c1 {
  margin-top: 3.2rem;
}

.c2 {
  background-color: #FAFAFA;
  color: #030517;
  padding: 1.6rem;
}

@media (min-width:768px) {
  .c1 {
    display: grid;
    grid-template-columns: 32rem 1fr;
    gap: calc(3.2rem * 2);
  }
}

<div
  data-testid="Mock Base"
>
  <div
    class="c0"
  >
    <div
      data-testid="Mock Heading"
    >
      My profile
    </div>
    <main
      class="c1"
    >
      <div
        data-testid="Mock ProfileMenu"
      />
      <div
        class="c2"
      >
        Lorem ipsum
      </div>
    </main>
  </div>
</div>
`;
